### PR TITLE
[#7651] Make use of msiSendMail an opt-in (4-3-stable)

### PR DESCRIPTION
**Pull request by korydraughn**
Cherry-pick of #7662

### CODE REVIEW

File: scripts/irods/test/test_all_rules.py [hunk 1]
```diff
@@ -275,6 +279,26 @@ def test(self):
                                                      ,'STDOUT_SINGLELINE', "completed successfully")
                 return test
 
+            # Returns a test with the following name:
+            #
+            #     test_<filename>_r
+            #
+            # Dots (.) are replaced with underscores (_).
```
Comment (trel) [inline scripts/irods/test/test_all_rules.py:286]: consider....    'Dots' -> 'Periods'

Comment (korydraughn) [reply scripts/irods/test/test_all_rules.py]: Done.

**Review by trel — APPROVED** [review]
i think this is ready if you're happy with it.

**Comment by korydraughn** [comment]
#'d

**Review by alanking — APPROVED** [review]

**Pull request merged**